### PR TITLE
Show error if target device or architecture files not specified.

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
### Motivate of the pull request
Show clear message to user that target device or architecture file was not specify.

### Describe the technical details
Check target device and architecture files before synth or place.
Show message in case architecture files were specify but unable to find them.

### CODE REVIEW

**Comment by KochynVolodymyr** [comment]
@alain-rs  please check if this is expected. We set default architecture file is main.cpp but it will fail if unable to find.

File: src/Compiler/CompilerOpenFPGA.cpp [hunk 1]
```diff
@@ -502,6 +502,14 @@ bool CompilerOpenFPGA::RegisterCommands(TclInterpreter* interp,
   return true;
 }
 
+bool CompilerOpenFPGA::VerifyTargetDevice() const {
+  const bool target = Compiler::VerifyTargetDevice();
+  const bool archFile = FileUtils::FileExists(m_architectureFile);
+  const bool openFpgaArchFile =
+      FileUtils::FileExists(m_OpenFpgaArchitectureFile);
+  return target || (archFile && openFpgaArchFile);
```
Comment (alaindargelas) [inline src/Compiler/CompilerOpenFPGA.cpp:510]: The test should be return (target || archFile);    for all the stages except bitstream stage.

The openFpgaArchFile is only needed for the bitstream generation stage.

Comment (KochynVolodymyr) [reply src/Compiler/CompilerOpenFPGA.cpp]: Ready

**Review by alaindargelas — APPROVED** [review]

**Comment by ravic-rs** [comment]
I am looking at how we can test these changes at foedag level with a tcl script, will merge after that.

**Review by ravic-rs — APPROVED** [review]
Looks good.

**Pull request merged**